brd-android/app-core/src/main/java/com/platform/tools/SessionHolder.kt

```diff
@@ -42,6 +42,15 @@ object SessionHolder : KodeinAware {
         return mApiSession
     }
 
+    @Synchronized
+    fun updateSessionState(state: SessionState): Session? {
+        if (mApiSession != null && mApiSession!!.state != state) {
```

Comment:
How about changing these lines for the following ones
```
mApiSession?.let {
   it.copy(state = state)
} ?:run {
    userManager.putSession(it)
}
```

It is not very idiomatic but much better than several `!!`